infrastructure/environments/mongodb.tf

```diff
@@ -0,0 +1,66 @@
+resource "azurerm_resource_group" "mongodb" {
+  location = var.location
+  name = format(local.name_format, "mongodb")
+
+  tags = {
+    app = local.app_name
+    env = terraform.workspace
+    lock = local.lock_delete
+  }
+}
+
+resource "random_integer" "mongodb" {
+  min = 1000
+  max = 9999
+}
+
+resource "azurerm_cosmosdb_account" "mongodb" {
+  name = format(local.name_format, "mongodb-${random_integer.mongodb.result}")
```

Comment:
not sure what's happening here honestly, but if there's any concern about two deploys ever hitting the same integer, could we potentially use a microsecond timestamp here instead?

Reply:
> tl;dr - good question, it's fine

The reason we need a pseudo-random integer is because Cosmos requires a unique name and it can take some time to register in Azure that the name is available - you can get into a weird state once deleting a resource and can't recreate it. By creating a random resource, this random number is linked to the workspace and only changes if the resource is destroyed (or tainted)